### PR TITLE
Adding of WebP support with PNG/JPEG as Fallback

### LINKED ISSUE #64: Use WebP Images with PNG Fallback to improve performance
To create a significant reduce of loading time, bandwidth, render time we can generate WebP Images out of PNG/JPEG on the Build Process.

To support Browser without WebP support (Safari) we can still add PNG/JPEG as Fallback.

### PR DESCRIPTION
**Pull request by dannylinden**
To resolve #64 here is the PR:

- Add WebP generation into the Build-Support
- Add src-webp url when necessary
- Update the image-component to create a picture set with Fallback for Non-WebP supporting Browser (Safari)

### CODE REVIEW

File: src/data/index_de.json [hunk 1]
```diff
@@ -198,7 +199,7 @@
                 "title": "Robert Koch Institut"
             }, {
                 "icon-url": "/assets/img/partner/sap_logo@1x.png",
-                "icon-set": "/assets/img/partner/sap_logo@1x.png 1x, /assets/img/partner/sap_logo@2x.png 2x",
+                "icon-set": "assets/img/partner/sap_logo@1x.png 1x, /assets/img/partner/sap_logo@2x.png 2x",
```
Comment (ayacoo) [inline src/data/index_de.json:202]: One slash too many away? :-)

File: src/data/index.json [hunk 2]
```diff
@@ -198,7 +200,7 @@
                 "title": "Robert Koch Institut"
             }, {
                 "icon-url": "/assets/img/partner/sap_logo@1x.png",
-                "icon-set": "/assets/img/partner/sap_logo@1x.png 1x, /assets/img/partner/sap_logo@2x.png 2x",
+                "icon-set": "assets/img/partner/sap_logo@1x.png 1x, /assets/img/partner/sap_logo@2x.png 2x",
```
Comment (ayacoo) [inline src/data/index.json:203]: One slash too many away?

File: src/data/index_de.json (hunk 1 above)
Comment (dannylinden) [reply src/data/index_de.json]: thx, fixed 👍

File: src/data/index.json (hunk 2 above)
Comment (dannylinden) [reply src/data/index.json]: thx, fixed 👍

**Comment by dannylinden** [comment]
- Slashes fixed
- .DS_STORE files removed (also in the commit hostory)
- Rebase to latest master to resolve merge confllict

please review again :)

File: src/partials/image-component.html [hunk 3]
```diff
@@ -1 +1,13 @@
-<div class="image-block"><img {{> image-attributes image=image}}/></div>
\ No newline at end of file
+<div class="image-block">
+	{{#if image.url-webp}}
+		<picture>
+			<source srcset="{{image.url-webp}}" type="image/webp">
+		
+		<img src="{{image.url}}"{{#if image.url-set}} srcset="{{image.url-set}}"{{/if}}{{#if image.title}} title="{{image.title}}"{{#unless image.alt}} alt="{{image.title}}"{{/unless}}{{/if}}{{#if image.alt}} alt="{{image.alt}}"{{/if}} />
```
Comment (ingomc) [inline src/partials/image-component.html:6]: if there is no alt-Text, there should be a fallback to an empty alt-Attribute

Comment (dannylinden) [reply src/partials/image-component.html:6]: What exactly should be the Fallback? If there is no alt text available there is no value of any kind of string? Or you mean not adding an alt attribute at all in this case?

Comment (ingomc) [reply src/partials/image-component.html:6]: if theres no image.alt and no image.title -> `alt=""` this is the Fallback

Comment (dannylinden) [reply src/partials/image-component.html]: Understand, is done, see latest commit

**Comment by dannylinden** [comment]
Image component create now an empty alt attribute alt="" if title and alt is not defined. Following the W3.org recommendation: https://www.w3.org/WAI/tutorials/images/decorative/

please review again

File: gulpfile.js [hunk 4]
```diff
@@ -143,6 +143,8 @@ function javascript() {
 function images() {
   return gulp
     .src('src/assets/img/**/*')
+    .pipe(webp())
+    .pipe(gulp.dest(PATHS.dist + '/assets/img'))
```
Comment (tkowark) [inline gulpfile.js:147]: In my local tests, this only generates the webp files, not the minified png and jpg files. If I switch the order, only png and jpg are created but not webp

Comment (tkowark) [reply gulpfile.js:147]: creating a separate function and adding it separately to the series seems to work, though

Comment (dannylinden) [reply gulpfile.js]: Good point, i will provide the fix

**Comment by dannylinden** [comment]
I fixed the pipeline, now always img (minify) and webp are generated. I also made a rebase to latest master. Please review again @tkowark

File: package-lock.json [hunk 5]
```diff
@@ -3925,12 +3897,12 @@
...
           "version": "3.0.0",
           "resolved": "https://registry.npmjs.org/get-stream/-/get-stream-3.0.0.tgz",
-          "integrity": "sha1-jpQ9E1jcN1VQVOy+LtsFqhdO3hQ=",
-          "optional": true
+          "integrity": "sha1-jpQ9E1jcN1VQVOy+LtsFqhdO3hQ="
         },
         "pify": {
           "version": "3.0.0",
           "resolved": "https://registry.npmjs.org/pify/-/pify-3.0.0.tgz",
+<<<<<<< HEAD
```
Comment (tkowark) [inline package-lock.json:3905]: There seems to be a merge conflict here.

Comment (dannylinden) [reply package-lock.json]: weird, github show now conflict issue. but i recreated them after rebase to master, should be fine now

**Comment by tkowark** [comment]
Libpng seems to be missing on the self-hosted runner. I'll resolve this soon and then revert the revert ...sorry about that and thank you again for the great work!

**Pull request merged**